nano/node/bootstrap/bootstrap_server.cpp

```diff
@@ -142,16 +179,44 @@ void nano::bootstrap_server::respond (nano::asc_pull_ack & response, std::shared
 	nano::transport::buffer_drop_policy::limiter, nano::transport::traffic_type::bootstrap);
 }
 
-/*
- * Requests
- */
+void nano::bootstrap_server::run ()
+{
+	nano::unique_lock<nano::mutex> lock{ mutex };
+	while (!stopped)
+	{
+		if (!queue.empty ())
+		{
+			stats.inc (nano::stat::type::bootstrap_server, nano::stat::detail::loop);
 
-void nano::bootstrap_server::process_batch (std::deque<request_t> & batch)
+			run_batch (lock);
```

Comment:
If batch was given to `run_batch` we would not have to give it the lock, which is a big ugly.

Reply:
Passing batch is arguably uglier, as it requires defining `run_batch(std::deque<decltype(queue)::value_type>)` with `queue` field having to be declared before the method.